packages/components/pie-card/src/index.ts

```diff
@@ -117,6 +120,57 @@ export class PieCard extends LitElement implements CardProps {
         return `padding: ${paddingCSS}`;
     }
 
+    /**
+     * Handles the slot change event and applies/removes opacity to images based on the `disabled` state.
+     *
+     * @private
+     */
+    private handleSlotChange () {
+        this.updateImagesOpacity();
+    }
+
+    /**
+     * Updates opacity of all images (slotted and non-slotted) based on the `disabled` property.
+     *
+     * @private
+     */
+    private updateImagesOpacity (): void {
+        if (this.assignedElements) {
+            // Handle slotted images
```

Comment:
I think this wording might be slightly clearer
```suggestion
            // Handle images nested inside slotted elements
```